bindings/pyroot/cppyy/CPyCppyy/src/Pythonize.cxx

```diff
@@ -841,6 +841,16 @@ PyObject* name##StringCompare(PyObject* self, PyObject* obj)                 \
 CPPYY_IMPL_STRING_PYTHONIZATION_CMP(std::string, Stl)
 CPPYY_IMPL_STRING_PYTHONIZATION_CMP(std::wstring, StlW)
 
+Py_hash_t StlStringHash(PyObject* self)
+{
+// std::string objects hash to the same values as Python strings to allow
+// matches in dictionaries etc.
+    PyObject* data = StlStringGetData(self);
+    Py_hash_t h = CPyCppyy_PyText_Type.tp_hash(data);
```

Comment:
Is `tp_hash` taking a reference?

Reply:
No I don't think so, see [this](https://docs.python.org/3/c-api/typeobj.html#c.PyTypeObject.tp_hash). It's just returning the hash of the pyobject.